core/server/master/src/main/java/alluxio/master/file/contexts/OperationContext.java

```diff
@@ -23,41 +29,50 @@
   // Proto message that is being wrapped
   private T mOptionsBuilder;
   // Used to track client call status.
-  private CallTracker mCallTracker;
+  private List<CallTracker> mCallTrackers;
 
   /**
    * Creates an instance with given proto message.
    *
    * @param optionsBuilder Internal proto message builder instance
    */
   public OperationContext(T optionsBuilder) {
-    this(optionsBuilder, null);
     mOptionsBuilder = optionsBuilder;
-    mCallTracker = CallTracker.DISABLED_TRACKER;
+    mCallTrackers = new LinkedList<>();
   }
 
   /**
-   * Creates an instance with given proto message.
-   *
-   * @param optionsBuilder Internal proto message builder instance
-   * @param callTracker client call tracker, or {@code null} if no tracking is desired
+   * @return underlying proto message instance
    */
-  public OperationContext(T optionsBuilder, CallTracker callTracker) {
-    mOptionsBuilder = optionsBuilder;
-    mCallTracker = callTracker;
+  public T getOptions() {
+    return mOptionsBuilder;
   }
 
   /**
-   * @return underlying proto message instance
+   * Adds a new call-tracker to this context.
+   *
+   * @param callTracker the call tracker
    */
-  public T getOptions() {
-    return mOptionsBuilder;
+  public void addCallTracker(CallTracker callTracker) {
+    synchronized (mCallTrackers) {
+      mCallTrackers.add(Preconditions.checkNotNull(callTracker));
```

Comment:
We should specify the parameter name:

```suggestion
      mCallTrackers.add(Preconditions.checkNotNull(callTracker, "callTracker"));
```

Reply:
n/a